framework/packages/abstract-client/src/client.rs

```diff
@@ -119,3 +120,56 @@ impl<Chain: MutCwEnv> AbstractClient<Chain> {
             .map_err(Into::into)
     }
 }
+
+pub mod daemon {
+    use abstract_core::{objects::AccountId, MANAGER};
+    use abstract_interface::AbstractAccount;
+    use cw_orch::{
+        daemon::Daemon,
+        environment::TxHandler as _,
+        state::{ChainState, StateInterface},
+    };
+
+    use super::*;
+
+    impl AbstractClient<Daemon> {
+        /// Retrieve the last account created by the client.
+        /// Returns `None` if no account has been created yet.
+        pub fn get_last_account(&self) -> AbstractClientResult<Option<Account<Daemon>>> {
+            let addresses = self.environment().state().get_all_addresses()?;
+
+            // Now search for all the keys that start with "abstract:manager-x" and return the one which has the highest x.
+            let mut last_account: Option<(u32, Account<Daemon>)> = None;
+            for (id, _) in addresses {
+                if !id.starts_with(MANAGER) {
+                    continue;
```

Comment:
Actually, can you cut it into different method, that takes `HashMap<String, Addr>` so it's unit-testable?

Reply:
yes :)

Reply:
Actually this will be hard to test because we need an actual Daemon object with an owner and we can't easily test all the different edge cases.

Reply:
> Actually this will be hard to test because we need an actual Daemon object with an owner and we can't easily test all the different edge cases.

Hmm, is it tho? Can't that helper method can take `CwEnv` instead and test against Mock?